.github/workflows/build-binaries-and-push-to-r2.yaml

```diff
@@ -0,0 +1,64 @@
+name: Upload core binary to Cloudflare R2 bucket
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: Github branch to checkout for build
+        required: true
+        default: main
+        type: string
+
+jobs:
+    build-and-upload:
+        name: Build binary and push to R2
+        runs-on: ubuntu-22.04
+        permissions:
+            contents: write
+            packages: write
+            id-token: write
+        steps:
+            - name: Determine branch
+              id: get-branch-name
+              run: |
+                if [ "${{ github.event_name }}" == "push" ]; then
+                  branch="main"
+                else
+                  branch="${{ inputs.branch }}"
+                fi
+                echo "branch=$branch" >> $GITHUB_OUTPUT
+
+            - name: Checkout code
+              uses: actions/checkout@v4
+              with:
+                fetch-depth: "0"
+                path: axelar-core
+                submodules: recursive
+                ref: ${{ steps.get-branch-name.outputs.branch }}
+            
+            - name: Build axelard binary with version and commit hash
+              id: build-binary
+              run: |
+                mkdir ./binary_file/
+                cd axelar-core
+                make build
```

Comment:
Use the same build makefile action
https://github.com/axelarnetwork/axelar-core/blob/main/.github/workflows/build-docker-image-and-binaries.yaml#L78

Reply:
Marking as resolved based on the conversation we had. Since we are using this action's result in a private repo, `make build` on linux amd64 would suffice for now.